ports/servoshell/desktop/webview.rs

```diff
@@ -88,6 +89,8 @@ pub struct WebView {
     pub url: Option<ServoUrl>,
     pub focused: bool,
     pub load_status: LoadStatus,
+    file_dialog: Option<FileDialog>,
+    file_response_sender: Option<IpcSender<Option<Vec<String>>>>,
```

Comment:
So we would like to support more than one dialog per WebView, I think, even if one is only shown at once. This would be represented a bit like this:

```rust
dialogs: Vec<Dialog>,
```

Then `Dialog` would be an enum like this:

```rust
enum Dialog {
    File(FileDialog),
}
```

`FileDialog` is a struct:

```rust
struct FileDialog {
    dialog: FileDialog,
    multiple: bool,
    response_sender: IpcSender<Option<Vec<String>>>,
}
```

I would put this new struct and enum into a file called `dialog.rs`.